spec/services/dqt_api_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe DqtApi do
+  describe '.find_trn!', vcr: true do
+    subject(:find_trn!) { described_class.find_trn!(trn_request) }
+
+    let(:trn_request) do
+      TrnRequest.new(date_of_birth: '1990-01-01', email: 'kevin@kevin.com', first_name: 'kevin', ni_number: '1000000')
+    end
+
+    it { is_expected.to match(hash_including('trn' => '1275362')) }
+    it { is_expected.to match(hash_including('emailAddresses' => ['anonymous@anonymousdomain.org.net.co.uk'])) }
+    it { is_expected.to match(hash_including('firstName' => 'Kevin')) }
+    it { is_expected.to match(hash_including('lastName' => 'Evans')) }
+    it { is_expected.to match(hash_including('dateOfBirth' => '1990-01-01')) }
+    it { is_expected.to match(hash_including('nationalInsuranceNumber' => 'SD78771P9')) }
+    it { is_expected.to match(hash_including('uid' => 'f7891223-7661-e411-8047-005056822391')) }
+
+    context 'when the API returns a timeout error' do
```

Comment:
Hmm. I don't think we should test this functionality by using the `FeatureFlag`, as we're not exercising the path that the "real code" will take. (And if we were to use a coverage tool, it would highlight an untouched branch in the controller)

Could we instead force the API to fail in this spec using a mock?

Reply:
I'll take a look at mocking a timeout